Kokkos::vector'is deprecated

I started to see this warning since I upgraded Trilinos:
```
In file included from /home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.cpp:9:
/home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.hpp:135:24: warning: 'template<class Scalar, class Arg1Type> class Kokkos::vector' is deprecated [-Wdeprecated-declarations]
  135 |     using KV = Kokkos::vector<DynRankView, PHX::Device>;
      |                        ^~~~~~
In file included from /home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.hpp:23,
                 from /home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.cpp:9:
/storage/workdir/trilinos/trilinos-install/groppello/gcc/opt/develop-serial/include/Kokkos_Vector.hpp:52:25: note: declared here
   52 | class KOKKOS_DEPRECATED vector
```
While I don't expect the class to disappear very soon, we should be proactive, and remove its usage. We don't want to forget and have to rush it when Kokkos finally removes it.

--- a/src/evaluators/gather/PHAL_GatherSolution.hpp
+++ b/src/evaluators/gather/PHAL_GatherSolution.hpp
@@ -20,8 +20,6 @@
 
 #include "Teuchos_ParameterList.hpp"
 
-#include "Kokkos_Vector.hpp"
-
 namespace Albany {
   class DOFManager;
 }
@@ -132,7 +130,7 @@ protected:
 public:
   struct SolAccessor {
     using DynRankView = Kokkos::DynRankView<ScalarT, PHX::Device>;
-    using KV = Kokkos::vector<DynRankView, PHX::Device>;
+    using KV = Kokkos::DualView<DynRankView*, PHX::Device>;
     using t_dev = typename KV::t_dev;
     
     KOKKOS_INLINE_FUNCTION

--- a/src/evaluators/gather/PHAL_GatherSolutionSide.hpp
+++ b/src/evaluators/gather/PHAL_GatherSolutionSide.hpp
@@ -19,8 +19,6 @@
 
 #include "Teuchos_ParameterList.hpp"
 
-#include "Kokkos_Vector.hpp"
-
 namespace Albany {
   class DOFManager;
 }

--- a/src/evaluators/gather/PHAL_GatherSolution_Def.hpp
+++ b/src/evaluators/gather/PHAL_GatherSolution_Def.hpp
@@ -170,7 +170,7 @@ postRegistrationSetup(typename Traits::SetupData d,
     }
     numNodes = val[0].extent(1);
 
-    // Get MDField views from std::vector
+    // Copy kokkos views from std::vector of MDFields to DualView of DynRankView
     for (int i =0; i<numFields;i++){
       device_sol.val_kokkos.h_view(i)=this->val[i].get_static_view();
       if (enableTransient){

--- a/src/evaluators/gather/PHAL_GatherSolution_Def.hpp
+++ b/src/evaluators/gather/PHAL_GatherSolution_Def.hpp
@@ -172,20 +172,23 @@ postRegistrationSetup(typename Traits::SetupData d,
 
     // Get MDField views from std::vector
     for (int i =0; i<numFields;i++){
-      device_sol.val_kokkos[i]=this->val[i].get_static_view();
+      device_sol.val_kokkos.h_view(i)=this->val[i].get_static_view();
       if (enableTransient){
-        device_sol.val_dot_kokkos[i]=this->val_dot[i].get_static_view();
+        device_sol.val_dot_kokkos.h_view(i)=this->val_dot[i].get_static_view();
       }
       if (enableAcceleration){
-        device_sol.val_dotdot_kokkos[i]=this->val_dotdot[i].get_static_view();
+        device_sol.val_dotdot_kokkos.h_view(i)=this->val_dotdot[i].get_static_view();
       }
     }
-    device_sol.val_kokkos.host_to_device();
+    device_sol.val_kokkos.modify_host();
+    device_sol.val_kokkos.sync_device();
     if (enableTransient){
-      device_sol.val_dot_kokkos.host_to_device();
+      device_sol.val_dot_kokkos.modify_host();
+      device_sol.val_dot_kokkos.sync_device();
     }
     if (enableAcceleration){
-      device_sol.val_dotdot_kokkos.host_to_device();
+      device_sol.val_dotdot_kokkos.modify_host();
+      device_sol.val_dotdot_kokkos.sync_device();
     }
 
   } else if (tensorRank == 1) {

--- a/src/evaluators/scatter/PHAL_ScatterResidual.hpp
+++ b/src/evaluators/scatter/PHAL_ScatterResidual.hpp
@@ -23,8 +23,6 @@
 
 #include "Teuchos_ParameterList.hpp"
 
-#include "Kokkos_Vector.hpp"
-
 namespace Albany {
 class DOFManager;
 }
@@ -98,7 +96,7 @@ protected:
 public:
   struct ResidAccessor {
     using DynRankView = Kokkos::DynRankView<const ScalarT, PHX::Device>;
-    using KV = Kokkos::vector<DynRankView, PHX::Device>;
+    using KV = Kokkos::DualView<DynRankView*, PHX::Device>;
     using t_dev = typename KV::t_dev;
 
     KOKKOS_INLINE_FUNCTION

--- a/src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp
+++ b/src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp
@@ -100,7 +100,7 @@ postRegistrationSetup(typename Traits::SetupData d,
     }
 
     for (int eq =0; eq<numFields;eq++){
-      // Get MDField views from std::vector
+      // Copy kokkos views from std::vector of MDFields to DualView of DynRankView
       device_resid.val_kokkos.h_view(eq)=this->val[eq].get_static_view();
     }
     device_resid.val_kokkos.modify_host();

--- a/src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp
+++ b/src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp
@@ -101,9 +101,10 @@ postRegistrationSetup(typename Traits::SetupData d,
 
     for (int eq =0; eq<numFields;eq++){
       // Get MDField views from std::vector
-      device_resid.val_kokkos[eq]=this->val[eq].get_static_view();
+      device_resid.val_kokkos.h_view(eq)=this->val[eq].get_static_view();
     }
-    device_resid.val_kokkos.host_to_device();
+    device_resid.val_kokkos.modify_host();
+    device_resid.val_kokkos.sync_device();
 
     numNodes = val[0].extent(1);
   } else  if (tensorRank == 1) {

